### PR TITLE
fix: light-client start should never block

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

`light-client` start should never block. 

**Description**

Currently, calling `light-client` start twice in a row will wait forever.

```js
const client = ...
await client.start();
await client.start(); // Never returns
```

Simplify the wait logic by leveraging the fact that `runLoop` already awaits for `started` state.

### CODE REVIEW

File: packages/light-client/src/index.ts [hunk 1]
```diff
@@ -160,22 +160,10 @@ export class Lightclient {
...
-          resolve();
-        }
-      };
-      this.emitter.on(LightclientEvent.statusChange, lightclientStarted);
-    });
-
-    void this.runLoop();
-
-    return startPromise;
+    return this.runLoop();
```
Comment (nazarhussain) [inline packages/light-client/src/index.ts]: The problem I see in this approach is that normally user will use `await client.start()`, that will block the event loop forever. 

The syntax `void this.runLoop()` is a way to put the run loop in the background so the `await client.start()` does not block.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: packages/light-client/src/index.ts (hunk 1 above)
Comment (jeluard) [reply packages/light-client/src/index.ts]: Makes sense, so I guess the previous approach made more sense.

Comment (nflaig) [reply packages/light-client/src/index.ts]: Can't we just check if light client is already started and return right away when `start()` is called? The second start call should be a no-op

**Review by nazarhussain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.18.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.18.0) :tada:

**Pull request merged**